src/main/java/org/fogbowcloud/saps/engine/core/archiver/storage/swift/SwiftPermanentStorage.java

```diff
@@ -207,18 +222,91 @@ public boolean delete(SapsImage task) {
+        try {
+            List<String> fileNames = swiftAPIClient.listFiles(containerName, dir);
+            LOGGER.info("Files List: " + fileNames);
 
-        LOGGER.info("Files List: " + fileNames);
-
-        for (String file : fileNames) {
-            LOGGER.debug("Trying to delete file " + file + " from " + containerName);
-            swiftAPIClient.deleteFile(containerName, file);
+            for (String file : fileNames) {
+                LOGGER.debug("Trying to delete file " + file + " from " + containerName);
+                swiftAPIClient.deleteFile(containerName, file);
+            }
+        } catch (IOException e) {
+            LOGGER.error("Error while list files of path [" + dir + "] from Object Storage", e);
+            return false;
         }
 
         return true;
     }
 
+    @Override
+    public List<AccessLink> generateAccessLinks(String taskId) {
+        List<AccessLink> filesLinks = new ArrayList<>();
+        List<String> files = new ArrayList<>();
+
+        String tasksDir = properties.getProperty(SapsPropertiesConstants.PERMANENT_STORAGE_TASKS_DIR);
+        String[] dirs = {INPUTDOWNLOADING_DIR, PREPROCESSING_DIR, PROCESSING_DIR};
+
+        for (String dir : dirs) {
+            String dirPath = String.format(SWIFT_TASK_STAGE_DIR_PATTERN, tasksDir, taskId, dir);
+            try {
+                files.addAll(this.swiftAPIClient.listFiles(this.containerName, dirPath));
+            } catch (IOException e) {
+                LOGGER.error("Error while list files of path [" + dir + "] from Object Storage", e);
+            }
+        }
+
+        for(String filePath : files) {
+            String link;
+            try {
+                link = generateTempURL(filePath);
+                Path p = Paths.get(filePath);
+                String name = p.getParent().getFileName().toString() + "/" + p.getFileName().toString();
+                filesLinks.add(new AccessLink(name, link));
+            } catch (NoSuchAlgorithmException | InvalidKeyException e) {
+                LOGGER.error("Error while generate temp url to File Path [" + filePath + "]", e);
+            }
+
+        }
+        return filesLinks;
+    }
+
+    private String generateTempURL(String filePath)
+        throws NoSuchAlgorithmException, InvalidKeyException {
+
+        Formatter objectStoreFormatter = new Formatter();
+        objectStoreFormatter.format("%s\n%s\n%s", "GET", Long.MAX_VALUE, filePath);
+        String signature = calculateRFC2104HMAC(objectStoreFormatter.toString(), this.swiftKey);
+        objectStoreFormatter.close();
+
+        objectStoreFormatter = new Formatter();
+        objectStoreFormatter.format(TEMP_DIR_URL, filePath, signature, Long.MAX_VALUE);
+        String res = objectStoreFormatter.toString();
+        objectStoreFormatter.close();
+
+        return res;
+    }
+
+    private String calculateRFC2104HMAC(String data, String key)
```

Comment:
what is this RFC*?

Reply:
In this case I believe that is Request for Comments which specifies HMAC.

Reply:
I renamed it for `runHMACAlgorithm`.